core/src/block.rs

```diff
@@ -792,4 +792,58 @@ mod tests {
 
         Revalidate::revalidate::<true>(&valid_block, wsv).unwrap();
     }
+
+    #[test]
+    fn transactions_are_atomic() {
```

Comment:
This isn't exactly whqt the test is testing. Atomicity can be established only using formal verification or at the very least a proptest.

Reply:
`failbox_atomic_revert` maybe? There's also a successful transaction, which imposes the ordering.

Reply:
`failed_transaction_revert` maybe?

Reply:
`failed_transactions_revert`.